mpas_analysis/shared/climatology/climatology.py

```diff
@@ -862,7 +863,10 @@ def _setup_climatology_caching(ds, startYearClimo, endYearClimo,
 
         cacheIndex = len(cacheInfo)
         for year in years:
-            cacheIndices[yearsInDs == year] = cacheIndex
+            for month in monthValues:
+                mask = numpy.logical_and(yearsInDs == year,
+                                         monthsInDs == month)
+                cacheIndices[mask] = cacheIndex
```

Comment:
It's set to a value that will never be true for ` dsYear = ds.where(ds.cacheIndices == cacheIndex, drop=True)` below